twisted/internet/interfaces.py

```diff
@@ -2207,6 +2207,28 @@ def clientConnectionForTLS(tlsProtocol):
 
 
 
+class IProtocolNegotiationFactory(Interface):
```

Comment:
🚗 👋 The use of the term "Factory" in the name is sort of a hint that the thing which may provide this will *also* be a protocol factory.  But there's nothing about this interface which actually *requires* that, which makes it a bit awkward name-wise, since "protocol" in this name evokes *either* a bytestring (the thing that ALPN means) or an `IProtocol` provider (the thing that "protocol factory" means).

How about something like `IProtocolNegotiator`?

Also: I feel like this is missing a method.  How do you find out which protocol your peer negotiated, later?  Shouldn't something tell you?